addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -359,13 +490,188 @@ odoo.define('website_slides.quiz', function (require) {
+         * question.
+         * @private
+         */
+        _onCreateQuizClick: function () {
+            var $elem = this.$('.o_wslides_js_lesson_quiz_new_question');
+            this.$('.o_wslides_js_quiz_add').addClass('d-none');
+            new QuestionFormWidget(this, {
+                slideId: this.slide.id,
+                sequence: this.quiz.questionsCount + 1
+            }).appendTo($elem);
+        },
+
+        /**
+         * When clicking on the edit button of a question it
+         * initialize a new QuestionFormWidget with the existing
+         * question as inputs.
+         * @param ev
+         * @private
+         */
+        _onEditQuestionClick: function (ev) {
+            var $elem = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            var question = this._getQuestionDetails($elem);
+            new QuestionFormWidget(this, {
+                oldElement: $elem,
+                question: question,
+                slideId: this.slide.id,
+                sequence: question.sequence,
+                update: true
+            }).insertAfter($elem);
+            $elem.hide();
+        },
+
+        /**
+         * When clicking on the delete button of a question it
+         * toggles a modal to confirm the deletion
+         * @param ev
+         * @private
+         */
+        _onDeleteQuestionClick: function (ev) {
+            var question = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            new ConfirmationDialog(this, {
+                questionId: question.data('questionId'),
+                questionTitle: question.data('title')
+            }).open();
+        },
+
+        /**
+         * Displays the created Question at the correct place (after the last question or
+         * at the first place if there is no questions yet) It also displays the 'Add Question'
+         * button or open a new QuestionFormWidget if the user wants to immediately add another one.
+         *
+         * @param event
+         * @private
+         */
+        _displayCreatedQuestion: function (event) {
+            var $lastQuestion = this.$('.o_wslides_js_lesson_quiz_question:last');
+            if ($lastQuestion.length !== 0) {
+                $lastQuestion.after(event.data.view);
+            } else {
+                this.$el.prepend(event.data.view);
+            }
+            this.quiz.questionsCount++;
+            event.data.widget.destroy();
+            if (event.data.save_and_new) {
+                this._onCreateQuizClick();
+            } else {
+                this.$('.o_wslides_js_quiz_add_question').removeClass('d-none');
+            }
+        },
+
+        /**
+         * Display the updated question in place of the inputs.
+         * @param event
+         * @private
+         */
+        _displayUpdatedQuestion: function (event) {
+            var widget = event.data.widget;
+            widget.$oldElem.replaceWith(event.data.view);
```

Comment:
Even though it works, this is a bit confusing to read.
I think we can improve it several ways:

- Renaming $oldElem in the widget to $editedQuestion (because it is the edited question element, right?)
- When triggering_up the display_updated_question event, instead of just passing the widget, you should pass `$replacedQuestion: self.$editedQuestion` so here you can do `event.data.$editedQuestion.replaceWith(event.data.view)`
- Renaming "view" to something more meaningful like "newQuestionRenderedTemplate" (or something better if you have an idea?).
- Update this method's docstring to reflect what you do exactly, I don't think you still do something "in place of the inputs" anymore.

I know this is just mostly variable renaming, but I think it's important.